Tests/Packet++Test/Tests/InfiniBandTests.cpp

```diff
@@ -0,0 +1,144 @@
+		{
+		case pcpp::Ethernet:
+		{
+			// now let's get the Ethernet layer
+			auto* ethernetLayer = parsedPacket.getLayerOfType<pcpp::EthLayer>();
+			PTF_ASSERT_NOT_NULL(ethernetLayer);
+			break;
+		}
+		case pcpp::VLAN:
+		{
+			// now let's get the Vlan layer
+			auto* vlanLayer = parsedPacket.getLayerOfType<pcpp::VlanLayer>();
+			PTF_ASSERT_NOT_NULL(vlanLayer);
+			break;
+		}
+		case pcpp::IPv4:
+		{
+			// let's get the IPv4 layer
+			auto* ipLayer = parsedPacket.getLayerOfType<pcpp::IPv4Layer>();
+			PTF_ASSERT_NOT_NULL(ipLayer);
+			break;
+		}
+		case pcpp::UDP:
+		{
+			// let's get the UDP layer
+			auto* udpLayer = parsedPacket.getLayerOfType<pcpp::UdpLayer>();
+			PTF_ASSERT_NOT_NULL(udpLayer);
+			PTF_ASSERT_EQUAL(udpLayer->getSrcPort(), 57236);
+			PTF_ASSERT_EQUAL(udpLayer->getDstPort(), 4791);
+			break;
+		}
+		case pcpp::InfiniBand:
+		{
+			// let's get the InfiniBand layer
+			auto* ibLayer = parsedPacket.getLayerOfType<pcpp::InfiniBandLayer>();
+			PTF_ASSERT_NOT_NULL(ibLayer);
+			PTF_ASSERT_EQUAL(ibLayer->getOpcode(), 12);
+			PTF_ASSERT_EQUAL(ibLayer->getSoliciteEvent(), false);
+			PTF_ASSERT_EQUAL(ibLayer->getMigrationState(), false);
+			PTF_ASSERT_EQUAL(ibLayer->getPadCount(), 0);
+			PTF_ASSERT_EQUAL(ibLayer->getTransportHeaderVersion(), 0);
+			PTF_ASSERT_EQUAL(ibLayer->getPartitionKey(), 65535);
+			PTF_ASSERT_EQUAL(ibLayer->getQueuePairNumber(), 17);
+			PTF_ASSERT_EQUAL(ibLayer->getFecn(), false);
+			PTF_ASSERT_EQUAL(ibLayer->getBecn(), false);
+			PTF_ASSERT_EQUAL(ibLayer->getAck(), 1);
+			PTF_ASSERT_EQUAL(ibLayer->getPacketSequenceNumber(), 5557091);
+			break;
+		}
+		case pcpp::GenericPayload:
+		{
+			break;
+		}
+		default:
+		{
+			break;
+		}
+		}
+	}
+}  // InfiniBandParsingTest
+
+PTF_TEST_CASE(InfiniBandCreationTest)
+{
+	pcpp::EthLayer ethLayer1(pcpp::MacAddress("02:7d:fa:01:17:40"), pcpp::MacAddress("02:7d:fa:00:10:01"),
+	                         PCPP_ETHERTYPE_IP);
+	pcpp::IPv4Layer ipLayer1(pcpp::IPv4Address("192.168.0.1"), pcpp::IPv4Address("192.168.0.2"));
+	pcpp::UdpLayer udpLayer1(30502, 4791);
+	pcpp::InfiniBandLayer infinibandLayer1(12, 0, 0, 0, 65535, 17, 1, 5557091);
+
+	pcpp::Packet infinibandPacket1(100);
+	PTF_ASSERT_TRUE(infinibandPacket1.addLayer(&ethLayer1));
+	PTF_ASSERT_TRUE(infinibandPacket1.addLayer(&ipLayer1));
+	PTF_ASSERT_TRUE(infinibandPacket1.addLayer(&udpLayer1));
+	PTF_ASSERT_TRUE(infinibandPacket1.addLayer(&infinibandLayer1));
+
+	PTF_ASSERT_EQUAL(infinibandPacket1.getLayerOfType<pcpp::UdpLayer>()->getDataLen(), 20);
+	PTF_ASSERT_NOT_NULL(infinibandPacket1.getLayerOfType<pcpp::InfiniBandLayer>());
+	PTF_ASSERT_EQUAL(udpLayer1.getDstPort(), 4791);
+	PTF_ASSERT_EQUAL(infinibandLayer1.getOpcode(), 12);
+	PTF_ASSERT_EQUAL(infinibandLayer1.getPartitionKey(), 65535);
+	PTF_ASSERT_EQUAL(infinibandLayer1.getQueuePairNumber(), 17);
+	PTF_ASSERT_EQUAL(infinibandLayer1.getAck(), 1);
+	PTF_ASSERT_EQUAL(infinibandLayer1.getPacketSequenceNumber(), 5557091);
```

Comment:
You can write a test like this: https://github.com/seladb/PcapPlusPlus/blob/master/Tests/Packet%2B%2BTest/Tests/DhcpV6Tests.cpp#L123-L124

Basically what it does is:
1. Create the InfiniBand layer
2. Load the packet from the `InfiniBand.dat` file and get the InfiniBand layer
3. Assert that the raw buffer length of the layer in (1) is equal to the raw buffer length of the layer in (2)
4. Assert that the raw buffer of the layer in (1) is equal to the raw buffer of the layer in (2) using `PTF_ASSERT_BUF_COMPARE`